File: Seminar/src/main/java/org/sopt/Seminar/domain/post/dto/PostGetResponse.java
package org.sopt.Seminar.domain.post.dto;

import org.sopt.Seminar.domain.category.model.Category;
import org.sopt.Seminar.domain.post.model.Post;

public record PostGetResponse(
        Long id,
        String title,
        String content,
        String category
) {
    public static PostGetResponse of(Post post, Category category) {
        return new PostGetResponse(
                post.getId(),
                post.getTitle(),
                post.getContent(),
                category.getContent()
        );
    }
}

File: kopring/week2/kotlin-semina/src/main/kotlin/org/sopt/kotlinsemina/domain/post/dto/PostGetResponse.kt
package org.sopt.kotlinsemina.domain.post.dto

import org.sopt.kotlinsemina.domain.category.model.Category
import org.sopt.kotlinsemina.domain.post.model.Post

data class PostGetResponse(
    val id: Long,
    val title: String,
    val content: String,
    val category: String,
) {
    companion object {
        fun of(post: Post, category: Category): PostGetResponse {
            return PostGetResponse(
                post.id,
                post.title,
                post.content,
                category.content)
        }
    }
}
